overlord/snapstate/handlers.go

```diff
@@ -208,16 +210,25 @@ func linkSnapInFlight(st *state.State, snapName string) (bool, error) {
 			if tc.Kind() == "link-snap" {
 				snapsup, err := TaskSnapSetup(tc)
 				if err != nil {
-					return false, err
+					return nil, err
 				}
 				if snapsup.InstanceName() == snapName {
-					return true, nil
+					return tc, nil
 				}
 			}
 		}
 	}
 
-	return false, nil
+	return nil, nil
+}
+
+func isLinkSnapInFlight(st *state.State, snapName string) (bool, error) {
```

Comment:
This helper does very little / isn't much different than just using findLinkSnapTask() directly, and appears to be used only once, so maybe just drop it and use findLinkSnapTask?